### ISSUE TITLE
The results of this program which I got was not right.

### ISSUE THREAD

**Issue by hli1221** [elleryqueenhomels/arbitrary_style_transfer#1]
Firstly, thank you very much. This program is very useful and easy to understand.
But after I train this network, I did not get right result. I just got an image with single color.
I change the BATCH_SIZE = 2, and use 10000 images(content and style) to train.
I do not know where I was wrong. 
I am very grateful if you have any advice. Thank you.

**Comment by elleryqueenhomels** [comment]
Hi Li! Thank you for your feedback. I really appreciate your finding. It is valuable for me. I have fixed the bug. You can git pull my latest version.

**Comment by hli1221** [comment]
Thank you for your answer. And I want to know, Can I run this code on CPU? 
Because when I run the latest version, the losses are all nan.

**Comment by elleryqueenhomels** [comment]
You can run the code on CPU, but it will be extremely slow. If you run it directly on CPU, you will get NaN. To address this problem, you need a tiny modification. In train.py, while computing the style loss, the equation "sigma = sqrt(var)", you need to add a small number epsilon => "sigma = sqrt(var + epsilon)".
BTW, I update my code just now, I replace the pkl file with a npz file which is converted from the torch model that is used by the author Xun Huang (https://github.com/xunhuang1995/AdaIN-style).

**Comment by hli1221** [comment]
Hi, my question still there. And I use 11000 images to train the network, the batch size is 4, epoch is 2. But still can not get the right result which is a single color image.
I wonder if you can send me your models of the decoder network.

**Comment by hli1221** [comment]
Or can you tell me the experimental setting? Like images number(80000?), batch size and the epochs. Can I get the right result if I fllow the setting in this program?

**Comment by elleryqueenhomels** [comment]
You should git pull my latest version to train the model. I am sure you can get a working model no matter which version (CPU / GPU) of TensorFlow you use, even though you use the setting "**BATCH_SIZE** = 4, **EPOCHS** = 2". If you want a trained model, please tell me your email address. However, my model only works, not perfect. Once I train a "perfect" model, I will upload it.

**Comment by elleryqueenhomels** [comment]
Currently, the best experimental setting is: **MS_COCO** number of images roughly **80k**; **WIkiArt** number of images roughly **80k**; **EPOCHS** >= 8 (I use 16, but 2 or 4 may also work); **BATCH_SIZE** = 8; **LEARNING_RATE** = 1e-4; Optimizer = **Adam**.
By the way, some images in the WikiArt **train** folder may throw warnings or exceptions while using **scipy.misc.imread**. A tradeoff solution is that you can delete these problematic images and replenish the WikiArt **train** folder with some images from WikiArt **test** folder. You should write a script to detect and remove these problematic images before you train the model. If you need my pre-process script I can send it to you.

**Comment by hli1221** [comment]
Thank you. I really appreciate your answer. I will try to run this code again. 
And I also want to get your trained model if it's convenient for you , my email is hui_li_jnu@163.com.
